examples/tutorials/text.py

```diff
@@ -0,0 +1,128 @@
+fig.coast(land="black", water="skyblue")
+
+# Plotting text annotations using single elements
+fig.text(text="SOUTH CHINA SEA", x=112, y=6)
+
+# Plotting text annotations using lists of elements
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+
+fig.show()
+
+########################################################################################
+# Changing font style
+# -------------------
+# The size, family/weight, and colour of an annotation can be specified using the ``font`` argument.
+#
+# A list of all recognised fonts can be found `here<https://docs.generic-mapping-tools.org/latest/cookbook/postscript_fonts.html>`_),
+# including details of how to use non-default fonts.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Customising the font style
+fig.text(text="BORNEO", x="114.", y=0.5, font="22p,Helvetica-Bold,white")
+
+fig.show()
+
+########################################################################################
+# Plotting from a text file
+# -------------------------
+#
+# It is also possible to add annotations from a file containing `x`, `y`, and
+# `text` fields. Here we give a complete example.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plot region names / sea names
+fig.text(textfiles="examples.txt", font="22p,Helvetica-Bold,white")
+
+# Plot names of seas
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+fig.text(text="SULU SEA", x=119.12, y=7.25, angle=-40)
+fig.text(text="SOUTH CHINA SEA", x=112, y=6, angle=40)
+fig.text(text="MAKASSAR STRAIT", x=118.4, y=-1, angle=65)
+fig.show()
+
+########################################################################################
+# ``justify`` argument
+# --------------------
+#
+# ``justify`` is used to define the anchor point for the bounding box for text
+# being added to a plot. The following code segment demonstrates the
+# positioning of the anchor point relative to the text.
+#
+# The anchor is specified with a two letter (order independent) code, chosen
+# from:
+# * Horizontal anchor: L(eft), C(entre), R(ight)
+# * Vertical anchor: T(op), M(iddle), B(ottom)
+
+fig = pygmt.Figure()
+fig.basemap(region=[0, 4, 0, 4], projection="X10c", frame="WSen")
+for i, y_justify in enumerate(["T", "M", "B"]):
+    y_pos = 3.5 - i*1.5
+    fig.plot(x=[0., 4.], y=[y_pos, y_pos], W='3p,red@85')
+    for j, x_justify in enumerate(["L", "C", "R"]):
+        justify_text = x_justify + y_justify
+        x_pos = 0.5 + j*1.5
+        fig.text(text=justify_text, x=x_pos, y=y_pos,
+                 font="28p,Helvetica-Bold,black", justify=justify_text)
+        fig.plot(x=[x_pos, x_pos], y=[0., 4.], W="3p,red@85")
+fig.show()
```

Comment:
I've added two commits in #481 that could simplify this section at 9e7a3534b2ba5f5668cdd94db1954a554308d6c0 and 187a733d72fc552284bdad3d5c2b81ad5c6d878d. Instead of passing an x/y pair, we could use `position="TR"` to plot text on the Top Right. Something like this:

```python
fig = pygmt.Figure()
fig.basemap(region=[0, 5, 0, 2.5], projection="x1c", frame="a")
for position in ("TL", "TR", "BL", "BR"):
    fig.text(position=position, text=position, offset="j0.5c+v")
fig.show()
```
produces:

![Four corner text plot](https://github.com/GenericMappingTools/pygmt/raw/187a733d72fc552284bdad3d5c2b81ad5c6d878d/pygmt/tests/baseline/test_text_position_offset_with_line.png)

Reply:
Ok, I've simplified the code at commit cb9f7137518de2433f2be86a1a22d0e0bb8ac72a.